components/camel-atlasmap/src/main/java/org/apache/camel/component/atlasmap/AtlasMapEndpoint.java

```diff
@@ -0,0 +1,444 @@
+    }
+
+    public String getTargetMapName() {
+        return this.targetMapName;
+    }
+
+    /**
+     * {@link TargetMapMode} enum value to specify how multiple target documents are delivered if exist.
+     * <ul>
+     * <li>'MAP': Stores them into a java.util.Map, and the java.util.Map is set to an exchange" property if
+     * 'targetMapName' is specified, otherwise message body.</li>"
+     * <li>'MESSAGE_HEADER': Stores them into message headers.</li>"
+     * <li>'EXCHANGE_PROPERTY': Stores them into exchange properties.</li>
+     * </ul>
+     * ")
+     * 
+     * @param mode {@link TargetMapMode}
+     */
+    public void setTargetMapMode(TargetMapMode mode) {
+        this.targetMapMode = mode;
+    }
+
+    public TargetMapMode getTargetMapMode() {
+        return this.targetMapMode;
+    }
+
+    public AtlasMapEndpoint findOrCreateEndpoint(String uri, String newResourceUri) {
+        String newUri = uri.replace(getResourceUri(), newResourceUri);
+        log.debug("Getting endpoint with URI: {}", newUri);
+        return getCamelContext().getEndpoint(newUri, AtlasMapEndpoint.class);
+    }
+
+    @Override
+    protected void onExchange(Exchange exchange) throws Exception {
+        Message incomingMessage = exchange.getIn();
+        String newResourceUri = incomingMessage.getHeader(AtlasMapConstants.ATLAS_RESOURCE_URI, String.class);
+        if (newResourceUri != null) {
+            incomingMessage.removeHeader(AtlasMapConstants.ATLAS_RESOURCE_URI);
+
+            log.debug("{} set to {} creating new endpoint to handle exchange", AtlasMapConstants.ATLAS_RESOURCE_URI,
+                    newResourceUri);
+            AtlasMapEndpoint newEndpoint = findOrCreateEndpoint(getEndpointUri(), newResourceUri);
+            newEndpoint.onExchange(exchange);
+            return;
+        }
+
+        AtlasSession atlasSession = getOrCreateAtlasContext(incomingMessage).createSession();
+        populateSourceDocuments(exchange, atlasSession);
+        atlasSession.getAtlasContext().process(atlasSession);
+
+        List<Audit> errors = new ArrayList<>();
+        for (Audit audit : atlasSession.getAudits().getAudit()) {
+            switch (audit.getStatus()) {
+                case ERROR:
+                    errors.add(audit);
+                    break;
+                case WARN:
+                    LOG.warn("{}: Document='{}(ID:{})', path='{}'",
+                            audit.getMessage(), audit.getDocName(), audit.getDocId(), audit.getPath());
+                    break;
+                default:
+                    LOG.info("{}: Document='{}(ID:{})', path='{}'",
+                            audit.getMessage(), audit.getDocName(), audit.getDocId(), audit.getPath());
+            }
+        }
+        if (!errors.isEmpty()) {
+            StringBuilder buf = new StringBuilder("Errors: ");
+            errors.stream().forEach(a -> buf.append(
+                    String.format("[%s: Document='{}(ID:{})', path='%s'], ",
```

Comment:
Do we have a mix here where last 2 params are ignored ?

Reply:
Doh good catch, thanks! fixed.